easybuild/tools/filetools.py

```diff
@@ -456,11 +456,14 @@ def extract_file(fn, dest, cmd=None, extra_options=None, overwrite=False, forced
     _log.debug("Unpacking %s in directory %s", fn, abs_dest)
     cwd = change_dir(abs_dest)
 
-    if not cmd:
-        cmd = extract_cmd(fn, overwrite=overwrite)
-    else:
+    if cmd:
         # complete command template with filename
         cmd = cmd % fn
+        _log.debug("Using specified command to unpack %s: %s", fn, cmd)
+    else:
+        cmd = extract_cmd(fn, overwrite=overwrite)
+        _log.debug("Using command derived from file extension to unpack %s: %s", fn, cmd)
+
     if not cmd:
         raise EasyBuildError("Can't extract file %s with unknown filetype", fn)
```

Comment:
if `extract_cmd()` fails, this error will not be shown, right? if so, better to try-except instead:

```suggestion
        try:
            cmd = extract_cmd(fn, overwrite=overwrite)
            _log.debug("Using command derived from file extension to unpack %s: %s", fn, cmd)
        except KeyError as err:
            raise EasyBuildError("Can't extract file %s with unknown filetype", fn)
```

Reply:
This `if not cmd` is a but superfluous, since the `file_extension` called in `extract_cmd` will already raise a meaningful error message (which I improved a bit in 1d4e45492).

I would like to keep this guard though, just in case we somehow accidentally make `extract_cmd` produce an empty extract command or something, it doesn't make sense to continue and let `eb` crash hard later with a cryptic error message in that case...